src/consensus/aft/impl/view_change_tracker.h

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "consensus/aft/raft_types.h"
+
+#include <chrono>
+#include <map>
+#include <set>
+
+namespace aft
+{
+  class ViewChangeTracker
+  {
+    struct ViewChange
```

Comment:
I'm not entirely clear on what this is. Is that keeping track of acks for a view change proposal effectively?

Reply:
It is keeping track of the actual messages (evidence). The way the new primary proves that it can be the new primary is that it sends the evidence to the other replicas and then puts said evidence onto the ledger when it is the primary. This is the final part of #1709

Reply:
Got it, as discussed I think it would be good to have those be called ViewChangeProposal, ViewChangeRequest, as opposed to ViewChangeComplete,  ViewChangeEvidence or whatever we end up using to refer to the transaction executed by the primary to record the completion of the view change to the ledger once enough valid ViewChangeProposal/Request/Votes have been gathered.

Reply:
I ended up going with RequestViewChange but same thing